### PR TITLE
[Coordination.Azure] Fix lease update and prune bug

### LINKED ISSUE #1251: [Lease.Azure] Unexpected cluster instability
**Version Information**
Version of Akka.NET? 1.4.48
Which Akka.NET Modules? Akka.Coordination.Azure 1.0.0-beta2

**Describe the bug**

After upgrading to these module versions:
Versions

Sharded cluster were unstable because the lease state were corrupted:
Exceptions

**Expected behavior**
Sharded cluster should be stable, lease should work as intended.

**Actual behavior**
Sharded cluster were unstable

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes #1251

## Changes

* Force "void" `PipeTo` to return `Status.Success` on successful invocation.

### CODE REVIEW

File: src/discovery/azure/Akka.Discovery.Azure/Actors/HeartbeatActor.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ protected override void OnReceive(object message)
                     if(_log.IsDebugEnabled)
                         _log.Debug("Updating cluster member entry TTL");
 
-                    ExecuteUpdateOpWithRetry().PipeTo(Self);
+                    ExecuteUpdateOpWithRetry().PipeTo(Self, success: () => Status.Success.Instance);
```
Comment (Aaronontheweb) [inline src/discovery/azure/Akka.Discovery.Azure/Actors/HeartbeatActor.cs]: What happens on failure?

**Review by Aaronontheweb — COMMENTED** [review]
Tests are failing, but also left a question

File: src/discovery/azure/Akka.Discovery.Azure/Actors/HeartbeatActor.cs (hunk 1 above)
Comment (Arkatufus) [reply src/discovery/azure/Akka.Discovery.Azure/Actors/HeartbeatActor.cs]: `PipeTo` always emits a `Status.Failure` on exceptions, unless you override it with a failure callback. The problem is with `PipeTo` that does not return anything (`async Task`, not `async Task<T>`), they're more akin to starting a detached `Task` that returns a `Status.Failure` when an exception was thrown

Comment (Aaronontheweb) [reply src/discovery/azure/Akka.Discovery.Azure/Actors/HeartbeatActor.cs]: That would do it. Got it, makes sense.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
```
   at Akka.Coordination.Azure.Internal.AzureApiImpl.LeaseResourceExists(String leaseName) in /home/vsts/work/1/s/src/coordination/azure/Akka.Coordination.Azure/Internal/AzureApiImpl.cs:line 227
   at Akka.Coordination.Azure.Internal.AzureApiImpl.ReadOrCreateLeaseResource(String name) in /home/vsts/work/1/s/src/coordination/azure/Akka.Coordination.Azure/Internal/AzureApiImpl.cs:line 59
   at Akka.Coordination.Azure.Tests.AzureApiSpec.ShouldUpdateLeaseConflict() in /home/vsts/work/1/s/src/coordination/azure/Akka.Coordination.Azure.Tests/AzureApiSpec.cs:line 80
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
```

Anything of note going on with this test failure @Arkatufus ?

**Comment by Arkatufus** [comment]
I think its the linux CI/CD image socket networking problem, the default connection timeout is 1 second, which I think a bit problematic on AzDo

**Comment by Arkatufus** [comment]
@Aaronontheweb https://github.com/akkadotnet/Akka.Management/pull/1315 should fix this, it bumps the default timeout value

**Pull request merged**